### ISSUE TITLE
[Bug] Pie chart roseType='area' incorrect when startAngle and endAngle given

### ISSUE THREAD

**Issue by MikkoSteerpath** [apache/echarts#19634]
### Version

5.5.0

### Link to Minimal Reproduction

https://echarts.apache.org/examples/en/editor.html?c=pie-roseType-simple&lang=ts&code=PYBwLglsB2AEC8sDeAoWsDOBTAThLGAXLANprrLkWxgCeIWxA5CPkwDRUUZgCGOYAILQA5gBtGsAIwAOAAztYXdFmgATYeMkBmAGwLlsAPRHYAWwjRNE4noPV0OYNgAq9SU35ZeHQ2t58xGQOFEiwAG68YgCuklJyCrDQvGYeABJRAGawgkywAL6cIehhkTGSAKyJyanMGWLZAEJ5hYYlEVGxxABMCYo16VmwAMIthgC6XPnkk_kA3EA

### Steps to Reproduce

Create a `roseType = 'area'` chart and set `startAngle` and `endAngle` so that `Math.abs(startAngle - endAngle) < Math.PI * 2`. Add at least two items.

### Current Behavior

Both the angles and radiuses of the pie chart's items are different.

### Expected Behavior

Only the radius of the items should be different, each item should have the same angle. Here is what the graph should look like https://echarts.apache.org/examples/en/editor.html?c=pie-roseType-simple&lang=ts&code=PYBwLglsB2AEC8sDeAoWsDOBTAThLGAXLANprrLkWxgCeIWxA5CPkwDRUUZgCGOYAILQA5gBtGsAIwAOAAztYXdFmgATYeMkBmAGwLlsALYRomicT0Hq6HMGwAVepKb8svDobW8-xMjYokWAA3XjEAV0kpOQVYaF4jFwAJMIAzWEEmWABfTgD0INCIyQBWWPjE5hSxdIAhLNzDApCwyOIAJhjFCuS02ABhBsMAXS5s8lHsgG4gA

### Any additional comments?

A workaround is to set minAngle to be some large number like 360 in which case it seems the fixed angle is respected again. This technique was used to get the "expected behavior".

**Comment by helgasoft** [comment]
> Both the angles and radiuses of the pie chart's items are different. Only the radius of the items should be different, each item should have the same angle.

Agree. And I'll throw a guess that ```roseType:'area'``` is not even implemented.  It behaves exactly like ```roseType:'radius'``` before being fixed by _minAngle_. BTW in order to work _minAngle_ needs to be (minAngle >= total angle / total sectors) .
![image](https://github.com/apache/echarts/assets/13038071/75e7ecc6-4a18-4005-9a93-e3d9a3497fa0)
